docs/faqs.md

```diff
@@ -0,0 +1,153 @@
+classical domain, and was chosen for *Jeff*.
+
+Human-readable formats have been quite common in the quantum space thus far, but in addition we see
+two major reason why they might make less sense going forward:
+
+- We are only just beginning to scale quantum software to FTQC regimes, so have not experienced the
+  performance issues associated with (re-)parsing human readable formats, compared to formats
+  designed for machine-readability.
+
+- the tooling around IR and exchange formats has not been mature enough, and the best way to
+  debug/inspect has often been to simply open and read the file. Improving tooling for debugging
+  and visualization should help rectify this.
+
+## Why do we need program structure?
+
+There are two main reasons:
+
+- Because we *need* to (not all algorithmic primitives, an example being repeat-until-success, can
+  be represented without structure!), and
+
+- Even when we don't *need* to, preserving structure is necessary for scalability.
+
+Quantum algorithms naturally have structure, even if we aren't used to thinking of them that way.
+For example, consider Shor's algorithm: when coded up in Python, we will instinctively use
+structure such as if statements, for loops, etc., rather than writing out the finalized list of
+gates to be applied. This structure results in a *compressed* program representation compared to
+the unrolled list of quantum instructions, which is both more efficient to compile, and may even
+allow for optimizations which may not be possible when the structure is lost/the program is flattened.
+
+So ideally, we should design compiler passes to act on
+such compressed representations! In extreme cases, if the program representation does not increase in size as the program scales,
+[compilation may be constant-time](https://arxiv.org/abs/2504.12449).
+
+There is still a lot of work to do when it comes to compiling structured programs:
+
+- Often, porting known state-of-the-art compilation techniques to apply to structured programs can
+  be highly non-trivial.
+
+- There is tension between maximal optimization and compact program representation. There are still
+  a lot of questions we need to consider and resolve when it comes to compiling dynamic, structured
+  programs. For example, the more compact the program representation (which is great for
+  compilation scalability!) the less information we potentially have up-front to maximally optimize
+  programs at compile time. How do we deal with such scenarios?
+
+## How did you decide what goes into the schema?
+
+## Why Cap'n Proto?
+
+We use [Cap'n Proto](https://capnproto.org/) as our data serialization format for Jeff.
+
+**Reasons for choosing Cap'n Proto:**
+- Fast, efficient binary encoding
+- Zero-copy deserialization for high performance
+- Simple and expressive schema language
+- Robust tooling and support for key quantum programming languages (Python, C++, Rust)
```

Comment:
```suggestion
- Robust tooling and support for key quantum programming frameworks (Python, C++, Rust)
```

I don't think the compilers/frameworks are formal languages, but tbh I don't know enough to explain precisely where the line is drawn.